spec/core/filtering/post_process_engine_spec.rb

```diff
@@ -0,0 +1,193 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'core/filtering'
+
+describe Core::Filtering::PostProcessEngine do
+  let(:job_description) do
+    {
+      Core::Filtering::FILTERING => [
+        {
+          Core::Filtering::DOMAIN => Core::Filtering::DEFAULT_DOMAIN,
+          Core::Filtering::RULES => rules,
+          Core::Filtering::ADVANCED_SNIPPET => snippet,
+          Core::Filtering::WARNINGS => []
+        }
+      ]
+    }
+  end
+  let(:rules) { [] }
+  let(:snippet) { {} }
+  let(:document) { { 'foo' => 'bar' } }
+  let(:test_field) { 'foo' }
+  let(:test_value) { 'bar' }
+  let(:test_rule_id) { 'test' }
+  let(:test_operator) { Core::Filtering::SimpleRule::Rule::EQUALS }
+  let(:test_rule) do
+    {
+      Core::Filtering::SimpleRule::ID => test_rule_id,
+      Core::Filtering::SimpleRule::FIELD => test_field,
+      Core::Filtering::SimpleRule::VALUE => test_value,
+      Core::Filtering::SimpleRule::POLICY => Core::Filtering::SimpleRule::Policy::EXCLUDE,
+      Core::Filtering::SimpleRule::RULE => test_operator
```

Comment:
You're not wrong. I constantized them because there were a few threads where the values seemed like they might be in-flux (`=` vs `equals` vs `Equals`, `<` vs `lt` vs `less_than`). I figured using constants would make it easier to refactor.

Reply:
I switched to using string literals for these keys